src/integrationTest/groovy/wooga/gradle/github/GithubPublishPropertySpec.groovy

```diff
@@ -338,6 +339,57 @@ class GithubPublishPropertySpec extends GithubPublishIntegration {
         methodName = useSetter ? "set${method.capitalize()}" : method
     }
 
+    @Issue("https://github.com/wooga/atlas-github/issues/31")
+    def "evaluates body property once"() {
+        given: "files to publish"
+        createTestAssetsToPublish(1)
+
+        and: "a buildfile with publish task"
+        buildFile << """
+            version "$versionName"
+
+            task testPublish(type:wooga.gradle.github.publish.tasks.GithubPublish) {
+                from "releaseAssets"
+                tagName = "$tagName"
+                repositoryName = "$testRepositoryName"
+                token = "$testUserToken"
+            }            
+        """.stripIndent()
+
+        buildFile << """
+            def callCounter = 0
+            
+            def publishBodyStrategy = new wooga.gradle.github.publish.PublishBodyStrategy() {
+                @Override
+                String getBody(org.kohsuke.github.GHRepository ghRepository) {
+                    callCounter = callCounter + 1
+                    return "evaluate body count: " + callCounter.toString()
+                }
+            }
+
+            testPublish.body(publishBodyStrategy)
+            
+            testPublish.doLast {
+                println("evaluate body count: " + callCounter.toString())
+            }
+        """.stripIndent()
+
+        when:
+        def result = runTasksSuccessfully("testPublish")
+
+        then:
+        hasReleaseByName(versionName)
+        def release = getReleaseByName(versionName)
+
+        release.getBody() == "evaluate body count: 1"
+        outputContains(result, "evaluate body count: 1")
```

Comment:
Maybe ensure no output with "evaluate body count: 2"

Reply:
I have to think about it. At the  moment I would leave it like this. I merged already the 2.x version with the same test.